### PR TITLE
Add .clang-format file

### PR DESCRIPTION
**Pull request by ericcurtin**
So people can use things like clang-format and git clang-format to style their code. Copied from ggml-cann, with PointerAlignment changed to Middle.

- [x] I have read the [contributing guidelines](https://github.com/ggerganov/llama.cpp/blob/master/CONTRIBUTING.md)
- Self-reported review complexity:
  - [x] Low
  - [ ] Medium
  - [ ] High

### CODE REVIEW

**Comment by ericcurtin** [comment]
Example of PR styled with this .clang-format file:

https://github.com/ggerganov/llama.cpp/pull/10291

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+BreakBeforeBinaryOperators: None
+BreakBeforeBraces: Attach
+BreakBeforeInheritanceComma: false
+BreakInheritanceList: BeforeColon
+BreakBeforeTernaryOperators: true
+BreakConstructorInitializersBeforeComma: false
+BreakConstructorInitializers: BeforeColon
+BreakAfterJavaFieldAnnotations: false
+BreakStringLiterals: true
+ColumnLimit:     80
```
Comment (ggerganov) [inline .clang-format:53]: I think most of the code uses 120 columns:

```suggestion
ColumnLimit:     120
```

Comment (ericcurtin) [reply .clang-format]: Cool, this is the first time I've seen that Commit suggestion button on GitHub, that's really handy!

Comment (ericcurtin) [reply .clang-format]: It does an automatic:

Co-authored-by: Georgi Gerganov <ggerganov@gmail.com>

which is sweet!

**Review by ggerganov — APPROVED** [review]
This should be a good starting point for `.clang-format` - we can iterate over it on `master`. @slaren wdyt?

**Comment by slaren** [comment]
I am not convinced that this is any better than starting from scratch. I may spend time on the next days trying to make one that we can use on every file, it would save a lot of time and fix all the minor inconsistencies that are present in nearly every PR from new contributors.

**Pull request closed without merging**